server/router/src/main/java/org/infinispan/server/router/configuration/builder/RestRouterBuilder.java

```diff
@@ -0,0 +1,37 @@
+package org.infinispan.server.router.configuration.builder;
+
+import java.util.Optional;
+
+import org.infinispan.server.router.configuration.RestRouterConfiguration;
+
+/**
+ * Configuration builder for REST.
+ *
+ * @author Sebastian Łaskawiec
+ */
+public class RestRouterBuilder extends AbstractRouterBuilder {
+
+    /**
+     * Creates new {@link RestRouterConfiguration}.
+     *
+     * @param parent Parent {@link MultiTenantConfigurationBuilderParent}.
+     */
+    public RestRouterBuilder(MultiTenantConfigurationBuilderParent parent) {
+        super(parent);
+    }
+
+    /**
+     * Builds {@link RestRouterConfiguration}.
+     */
+    public Optional<RestRouterConfiguration> build() {
+        if (this.enabled) {
+            try {
+                validate();
+            } catch (Exception e) {
+                throw logger.configurationValidationError(e);
+            }
+            return Optional.of(new RestRouterConfiguration(ip, port));
+        }
+        return Optional.empty();
```

Comment:
Maybe it would be ok if we started writing a new configuration from scratch, but when all the other configuration elements have an `enabled` field, I don't think it's worth breaking the convention.